azurelinuxagent/ga/exthandlers.py

```diff
@@ -1091,28 +1097,61 @@ def get_status_file_path(self, extension=None):
 
     def collect_ext_status(self, ext):
         self.logger.verbose("Collect extension status")
-
         seq_no, ext_status_file = self.get_status_file_path(ext)
         if seq_no == -1:
             return None
 
+        data = None
+        data_str = None
         ext_status = ExtensionStatus(seq_no=seq_no)
         try:
             data_str = fileutil.read_file(ext_status_file)
             data = json.loads(data_str)
-            parse_ext_status(ext_status, data)
         except IOError as e:
-            ext_status.message = u"Failed to get status file {0}".format(e)
-            ext_status.code = -1
-            ext_status.status = "error"
-        except ExtensionError as e:
-            ext_status.message = u"Malformed status file {0}".format(e)
-            ext_status.code = ExtensionErrorCodes.PluginSettingsStatusInvalid
-            ext_status.status = "error"
+            msg = u"Failed to find any status for extension - {0}-{1}, Sequence number {2}. Failed due to {3}" \
```

Comment:
Yes. When the file was unavailable, or we couldn't read it due to some permissions issue.

Reply:
> Then we need to improve "Failed to find any status"

I can improve the wording just a little to make the extension and the fault clearer. But apart from this, we don't have much context to be sent to the customer on what could be the failure, which could be preventing the extension to generate the status.
```
msg = u"We couldn't find any status for {0}-{1} extension, for the sequence number {2}. It failed due to {3}".format(ext.name, self.ext_handler.properties.version, seq_no, e)
```

> Also, it seems like those kind of errors are permanent, so we should fail the extension instead of setting it up to warning.

The need for making them warning came after an extension (backup or encryption) requested that we retry status reading or send non-terminal status because they couldn't generate the status file in time. Even if we want to fail them early, there have been cases that it was temporary.